src/Solver/Interoperability.cpp

```diff
@@ -655,13 +673,18 @@ void seissol::Interoperability::initializeModel(  char*   materialFileName,
         calcWaveSpeeds(&materials[i], i);
       }
     } 
+    delete queryGen;
 
     //now initialize the plasticity data
     if (plasticity) {
+      if (useMaterialAverage) {
+        logWarning() << "Material Averaging is not implemented for plastic materials. Use material sampled from the element barycenters instead.";
+      }
+      const seissol::initializers::ElementBarycentreGenerator queryGen(seissol::SeisSol::main.meshReader());
       auto materials = std::vector<seissol::model::Plasticity>(nElements);
       seissol::initializers::MaterialParameterDB<seissol::model::Plasticity> parameterDB;
       parameterDB.setMaterialVector(&materials);
-      parameterDB.evaluateModel(std::string(materialFileName), queryGen);
+      parameterDB.evaluateModel(std::string(materialFileName), &queryGen);
```

Comment:
delete queryGen at the end of the function?

Reply:
It is done in line 675

Reply:
but it is above this line?

Reply:
I supppose it is also a stack variable which does not need to be deleted?
(and L675 it is deleted to be able to use the same variable name?)

Reply:
I refactored it a little now.